### PR TITLE
Implement client/service package

### PR DESCRIPTION
**Pull request by supl**
This PR implements the `client/service` package that contains the following main interfaces:
- ClientService.RegisterCertificate
- ClientService.RegisterContract
- ClientService.ExecuteContract
- ClientService.ValidateLedger

The end-to-end test and example will be proposed in another PR.

### CODE REVIEW

File: client/service/client_service.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+// ClientService defines the interface of the client service.
+type ClientService struct {
+	clientConfig                config.ClientConfig
+	signer                      crypto.Signer
+	ledgerConnection            *grpc.ClientConn
+	ledgerPrivilegedConnection  *grpc.ClientConn
+	auditorConnection           *grpc.ClientConn
+	auditorPrivilegedConnection *grpc.ClientConn
+}
```
Comment (supl) [inline client/service/client_service.go:21]: This is the main structure.
All functions that implement the client interfaces are associated with this structure.

File: client/service/util.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	clientError "github.com/scalar-labs/scalardl-go-client-sdk/v3/client/error"
+	"github.com/scalar-labs/scalardl-go-client-sdk/v3/ledger/statuscode"
+	"github.com/scalar-labs/scalardl-go-client-sdk/v3/rpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/proto"
+)
+
+func getClientErrorFromTrailer(trailer metadata.MD) (err clientError.ClientError) {
```
Comment (supl) [inline client/service/util.go:11]: This function is an internal function used to retrieve Scalar DL Status Code from gRPC's metadata.

File: client/service/ledger_validation.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+	if endAge < startAge || startAge < 0 || endAge > JavaMaxIntValue {
+		return result, fmt.Errorf("invalid ages specified")
+	}
+
+	if signer, err = crypto.NewEcdsaSha256Signer([]byte(s.clientConfig.PrivateKey)); err != nil {
+		return
+	}
+
+	if s.clientConfig.IsAuditorEnabled && s.clientConfig.IsAuditorLinearizableValidationEnabled {
```
Comment (supl) [inline client/service/ledger_validation.go:66]: Here implements linearizable ledger validation

File: client/service/ledger_validation.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+		result.Code = statuscode.OK
+
+		if len(executed.Proofs) > 0 {
+			result.Proof = executed.Proofs[0]
+		}
+
+		if len(executed.AuditorProofs) > 0 {
+			result.AuditorProof = executed.AuditorProofs[0]
+		}
+	} else {
```
Comment (supl) [inline client/service/ledger_validation.go:93]: Here implements non-linearizable ledger validation

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

**Review by thongdk8 — APPROVED** [review]
LGTM! Thank you!

File: client/service/contract_execution.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+	trailer = metadata.MD{}
+	if responseFromLedger, err = ledger.ExecuteContract(context.Background(), request, grpc.Trailer(&trailer)); err != nil {
+		if trailer.Len() > 0 {
+			err = getClientErrorFromTrailer(trailer)
+		}
+
+		return
+	}
+
+	// send the execution validation reqeust to Auditor.k
```
Comment (feeblefakie) [inline client/service/contract_execution.go]: Auditor.k? just a typo?

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
(left one minor comment. Looks like just a typo)

File: client/service/contract_execution.go (hunk 5 above)
Comment (supl) [reply client/service/contract_execution.go]: Thank you.
I think it's because of the Vim binding.
(maybe somehow I wanna move the cursor up)

Comment (supl) [inline client/service/contract_execution.go]:
```suggestion
	// send the execution validation reqeust to Auditor.
```

**Pull request merged**